recipes/roaring/all/conanfile.py

```diff
@@ -29,28 +40,44 @@ def _source_subfolder(self):
     def _build_subfolder(self):
           return "build_subfolder"
 
+    @property
+    def _has_axv_support(self):
+          return self.settings.arch in ["x86", "x86_64"]
+
+    @property
+    def _has_neon_support(self):
+          return "arm" in self.settings.arch
+
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if not self._has_axv_support:
+            del self.options.avx
+        if not self._has_neon_support:
+            del self.options.neon
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-        if self.settings.compiler.cppstd:
-            check_min_cppstd(self, "11")
 
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "11")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "CRoaring-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
 
     def _configure_cmake(self):
         if self._cmake:
             return self._cmake
         self._cmake = CMake(self)
-        self._cmake.definitions['ENABLE_ROARING_TESTS'] = False
-        self._cmake.definitions['ROARING_BUILD_STATIC'] = not self.options.shared
+        self._cmake.definitions["ROARING_DISABLE_X64"] = not self.settings.arch in ["x86", "x86_64"]
```

Comment:
This whole set of checks should not be needed at all.

Reply:
(At least not for version 0.3 and up.)

Reply:
Ok, I've remove most options, to just avoid arch native in 0.2.66

Reply:
Note that the test package fails anyway for 0.3.2 with apple-clang 10